src/Cameras/Inputs/StubCameraPointersInput.ts

```diff
@@ -0,0 +1,236 @@
+import { Nullable } from "../../types";
```

Comment:
I would like to avoid having test files in the main code. Tests (and associated code) must be in the test folder

Reply:
yea, i tried but failed to get this working from within `tests/unit/babylon/src/`.
I can't `import` from there: `error TS1148: Cannot use imports, exports, or module augmentations when '--module' is 'none'.`

If i define the `StubCameraPointersInput` in `tests/unit/babylon/src/Cameras/babylon.pointerInput.tests.ts` it compiles fine but the `.js` file does not have `BABYLON` defined,
leading to `ReferenceError: BABYLON is not defined` at runtime.

Any advice on this class of issue?

Reply:
I would recommend use plain JS here

Reply:
> I would recommend use plain JS here
I presume you mean "plain JS"  in place of `StubCameraPointersInput.ts`?
How do i include files without module loading? (Apologies for the noob question. No real JS experience here.)

I'm not sure that would work anyway;
Because `BaseCameraPointersInput` is an `abstract` class we won't be able to use it directly, either from TS or JS.
The methods we want to test are `protected` so we can't just take an existing CameraInput and override it's methods.

I think the ideal solution here would be to build `StubCameraPointersInput.ts` outside the main BABYLON bundle and include it here....

Reply:
> How do i include files without module loading? (Apologies for the noob question. No real JS experience here.)

Never mind that question,
I can just remove the `"module": "none"` line in `tests/unit/tsconfig.json`.

Reply:
> I think the ideal solution here would be to build `StubCameraPointersInput.ts` outside the main BABYLON bundle and include it here....
Yes as long as it is not in the main codebase

Reply:
How i can achieve this?

Reply:
Well there's a solution that works, although very ugly.
Not the clean tests i was aiming for at all....

Have a look at `StubCameraInput()` in `babylon.pointerInput.tests.ts`.
It stubs out the methods of `ArcRotateCameraPointersInput()`.

anyways, sleepy time for me.
later!

Reply:
I wonder why not simply using the code generated by your initial TS class as the JS code to use in test. It will give you the scaffolding at least

Reply:
ok just seen the stub you created :) This is fine for me to be honest.

Also wondering if you could use scene.simualtePointerXXX ?

Reply:
> I wonder why not simply using the code generated by your initial TS class as the JS code to use in test. It will give you the scaffolding at least.
> StubCameraPointersInput.ts has generated a .js that could be the file to use in tests directly

Yes, this what i was trying.
i could not find a way to include all the generated .js that `StubCameraPointersInput` depends on.
I'm sure it's possible, just i do not understand how JavaScript includes work.
(i'm fine debugging a C or C++ linker. Linking .js is just weird....)

> ok just seen the stub you created :) This is fine for me to be honest.

Hmmm. i think it smells bad.
Unit tests should test code in isolation.
Since my Stub depends on `ArcRotateCameraPointersInput`, the unit tests for `BaseCameraPointersInput` are dependent on future modifications of `ArcRotateCameraPointersInput` not overriding `BaseCameraPointersInput` methods.

> Also wondering if you could use scene.simualtePointerXXX ?

I looked into that. It only covers a subset of the events i needed. (Cant remember which events exactly.) If this was my day job i'd extend the simualtePointer model but i ran out of motivation.

Reply:
Ok for now. wE can improve it later

Reply:
Also I want to apologize, I did not want to demotivate you :( It is just that I need to keep the code as unbloated as possible. I hope you understand :(

Reply:
ah, don't worry about that.
The whole point of this process is so we can discuss the best way of doing things. The fact we have differing opinions on some points is what makes it a discussion.

If no one was standing their ground on what gets included the project would not be the success it is. No apologies to me are required for that.

Communicating purely by text is hard when you don't know the person on the other end.
You can pretty much guarantee if you've offended me it's going to take more than comments about code and i'll let you know.

On a different tangent, here's the hacky little animation that got me playing with Babylon: https://mrdunk.github.io/karaokefox/dist/
Once i get this PR submitted i'll get that fox walking about his forest.

Reply:
I love it! And I'm glad I did not upset you as I appreciate your work a lot!